pkg/provider/branch/trunk/trunk.go

```diff
@@ -425,29 +437,37 @@ func (t *trunkENI) CreateAndAssociateBranchENIs(pod *v1.Pod, securityGroups []st
 	return newENIs, nil
 }
 
-// DeleteAllBranchENIs deletes all the branch ENIs associated with the trunk and all the ENIs present in the cool down
-// queue, this is the last API call to the the Trunk ENI before it is removed from cache
-func (t *trunkENI) DeleteAllBranchENIs() {
-	// Delete all the branch used by the pod on this trunk ENI
+// DisassociateAllBranchENIs removes association between all the branch ENIs with the trunk. The branch ENIs will be deleted by the node termination finalizer
+// This is the last API call to the the Trunk ENI before it is removed from cache
+func (t *trunkENI) DisassociateAllBranchENIs() {
+	// Disassociate all the branch used by the pod on this trunk ENI, it will be cleaned up by the finalizer routine
 	// Since after this call, the trunk will be removed from cache. No need to clean up its branch map
 	for _, podENIs := range t.uidToBranchENIMap {
 		for _, eni := range podENIs {
-			err := t.deleteENI(eni)
-			if err != nil {
-				// Just log, if the ENI still exists it can be removed by the dangling ENI cleaner routine
-				t.log.Error(err, "failed to delete eni", "eni id", eni.ID)
+			err := retry.OnError(
+				wait.Backoff{
+					Duration: time.Millisecond * 100,
+					Factor:   3.0,
+					Jitter:   0.1,
+					Steps:    7,
+					Cap:      time.Second * 10,
+				},
+				func(err error) bool {
+					if strings.Contains(err.Error(), ec2Errors.NotFoundAssociationID) {
+						// association is already deleted, do not retry
+						return false
+					}
+					return true
+				}, func() error {
+					return t.ec2ApiHelper.DisassociateTrunkInterface(&eni.AssociationID)
+				},
+			)
+			if err != nil && !strings.Contains(err.Error(), ec2Errors.NotFoundAssociationID) {
+				// Just log, if the ENI disassociation fails, it will be force deleted
+				t.log.Error(err, "failed to disassociate eni", "eni id", eni.ID)
```

Comment:
Should we also count this new API call success and error in metrics? Also I think latency metrics may help as well at least after we just add this call.

Reply:
Both error and latency metrics was already added in the previous PR [here](https://github.com/aws/amazon-vpc-resource-controller-k8s/blob/eni-cleanup/pkg/aws/ec2/api/wrapper.go#L820)